### ISSUE TITLE
test of walletFactory contract upgrade on top of upgrade-11

### ISSUE THREAD

**Issue by turadg** [Agoric/Agoric#8305]
## What is the Problem Being Solved?

https://github.com/Agoric/agoric-sdk/pull/8147 is in master. It was originally planned to out as an upgrade handler, but it will go out as a simple core proposal on top of current Mainnet. That's currently upgrade-11 which is in a release branch.

We need a test of walletFactory contract upgrade happening on top of that upgrade-11 state.

## Description of the Design / Test Plan / Upgrade Considerations

 - [x] Cherry-pick the necessary changes to the upgrade-11 / mainnet branch.
 - [x] update docker upgrade tests on mainnet branch to verify a publish bundle of wallet factory built from the local repo, and upgrade through core-eval would work
 - cut a github release that contains the bundle artifact, publish to mainnet
 - pass a core eval governance proposal referencing the bundle hash
 - update master branch docker upgrade tests to publish the artifact pulled from GH release, and upgrade

This issue covers the 1st two steps.

(Thanks @mhofman for help with the plan. --@dckc)

## Security / Scaling Considerations

Nothing new since #8156

**Comment by dckc** [comment]
## backport ran into git merge conflicts

In the [dc-sw-backport](https://github.com/Agoric/agoric-sdk/tree/dc-sw-backport) branch, I tried cherry-picking commits from...

 - #8071
 - #8147

but the latter ran into merge conflicts: the `actions.sh` changes pre-date moving stuff from upgrade-11 to upgrade-12.

## add layer between agoric-11 and agoric-12?

In [dc-sw-coreprop](https://github.com/Agoric/agoric-sdk/tree/dc-sw-coreprop) / #8308, I'm trying to use some of the nicer js testing stuff in master.

The `builders` package is not in the container in some approaches.

Another thing I ran into is:

```
11:57PM ERR UPGRADE "agoric-upgrade-11" NEEDED at height: 595:                              
panic: UPGRADE "agoric-upgrade-11" NEEDED at height: 595:                                                                                                                               
goroutine 38 [running]:                                                                     
github.com/cosmos/cosmos-sdk/x/upgrade.BeginBlocker
```

**Comment by dckc** [comment]
addressed in #8311
